pkg/repository/maintenance.go

```diff
@@ -156,9 +183,14 @@ func buildMaintenanceJob(m MaintenanceConfig, param provider.RepoParam, cli clie
 		},
 	}
 
-	if affinity := veleroutil.GetAffinityFromVeleroServer(deployment); affinity != nil {
-		job.Spec.Template.Spec.Affinity = affinity
+	var affinity *v1.Affinity
+	if config != nil && len(config.LoadAffinities) > 0 {
+		affinity = csiutil.ToSystemAffinity(config.LoadAffinities)
+	}
+	if affinity == nil {
```

Comment:
We don't need to do this necessarily:

- Velero server's resource pattern is very different from maintenance jobs', there is no help to inherit from it
- It keeps a clear API/UX to have the configuration explicitly from one single resource